### PR TITLE
Refactor the code base and move packages to internal

### PR DESCRIPTION
**Pull request by soumya-codes**
This PR hopefully is the final instalment of major refactoring PR's required as pre-requisite for moving DiceDB to multithreading.

This PR organizes the code into multiple packages within internal

Please DO NOT merge any further PRs till this PR is merged.
Please review the code changes thoroughly.
cc @JyotinderSingh @lucifercr07 @AshwinKul28 @pratikpandey21

### CODE REVIEW

File: internal/server/async_tcp.go [hunk 1]
```diff
@@ -5,20 +5,27 @@ import (
...
 	"errors"
 	"fmt"
+	"io"
 	"net"
+	"strings"
 	"sync"
 	"syscall"
 	"time"
 
+	eval2 "github.com/dicedb/dice/internal/eval"
```
Comment (lucifercr07) [inline internal/server/async_tcp.go:15]: `eval2` required due to some name collision?

File: internal/server/async_tcp.go [hunk 2]
```diff
@@ -286,13 +293,62 @@ func (s *AsyncServer) executeCommandToBuffer(redisCmd *cmd.RedisCmd, buf *bytes.
...
+			return nil, false, fmt.Errorf("empty command")
+		}
+
+		command := strings.ToUpper(tokens[0])
+		cmds = append(cmds, &cmd.RedisCmd{
+			Cmd:  command,
+			Args: tokens[1:],
+		})
+
+		if command == "ABORT" {
```
Comment (lucifercr07) [inline internal/server/async_tcp.go:330]: nit: Shall we extract this as const command as used at multiple places?

File: internal/shard/shard_thread.go [hunk 3]
```diff
@@ -7,12 +7,15 @@ import (
 	"sync"
 	"time"
 
+	eval2 "github.com/dicedb/dice/internal/eval"
```
Comment (lucifercr07) [inline internal/shard/shard_thread.go:10]: nit: Tried moving it locally to `"github.com/dicedb/dice/internal/eval"` server runs fine. Please let me know if I'm missing something.

**Review by lucifercr07 — APPROVED** [review]
Thanks @soumya-codes for the changes. Have very minor comments please have a look once. Approved.

File: internal/server/async_tcp.go (hunk 2 above)
Comment (soumya-codes) [reply internal/server/async_tcp.go:330]: Can i take these changes while working on the worker/multi-threading part? I wanted to limit the refactoring to code organization.

Comment (lucifercr07) [reply internal/server/async_tcp.go:330]: Sure just `nit`

File: internal/shard/shard_thread.go (hunk 3 above)
Comment (soumya-codes) [reply internal/shard/shard_thread.go:10]: no nothing, caused by some refactoring blues of moving stuff around...
thanks for catching

File: internal/server/async_tcp.go (hunk 2 above)
Comment (soumya-codes) [reply internal/server/async_tcp.go]: thanks

File: internal/server/async_tcp.go (hunk 1 above)
Comment (soumya-codes) [reply internal/server/async_tcp.go]: done

**Comment by AshwinKul28** [comment]
Thanks @soumya-codes LGTM, one more thought, shall we move all the commands to a single constant file? I see multiple occurrences of each command throughout the project! If you dont want to do that now, that's okay will take care of it once this PR is merged! Upto you!

**Review by JyotinderSingh — APPROVED** [review]
Thanks for the reorganization effort @soumya-codes! The changes look good.

**Pull request merged**